### PR TITLE
Feature/swagger docs

### PR DESCRIPTION
**Pull request by knordstrom**
Initial commit with swagger etc

### CODE REVIEW

File: library/api_models.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+from flask_restx import Api, fields
+from pydantic import AliasChoices, BaseModel, EmailStr, Field, ConfigDict
+
+import typing
+from typing import List, Optional, Union, TypeVar, Generic
+from dataclasses import dataclass, Field, fields as dataclassFields
+from flask_restx import fields as flaskRPFields
+
+
+
```
Comment (mshash8) [inline library/api_models.py:12]: Are these what you mean by strongly typed?

File: library/briefing_support.py [hunk 2]
```diff
@@ -71,11 +75,131 @@ def doc_context_for(event: dict, certainty: float = None) -> list[dict]:
...
+        filtered_data = list(filter(lambda x: x['email_id'] == email_id, data_list))
+        if filtered_data:
+            head = filtered_data[0]  # Get the first item
+            result = head['from']
+        else:
+            result = None
+        return result
+
+    @staticmethod
+    def construct_conversation_and_summary(emails_dict: dict) -> dict:
```
Comment (mshash8) [inline library/briefing_support.py]: I need to remove this comment. Some of my changes seem to be here too.

File: library/api_models.py (hunk 1 above)
Comment (knordstrom) [reply library/api_models.py:12]: Yes, pydantic lets you build well-typed models that won't have issues with indexing due to spelling or lack of visibility

Comment (mshash8) [reply library/api_models.py:12]: This is great actually!

**Pull request merged**